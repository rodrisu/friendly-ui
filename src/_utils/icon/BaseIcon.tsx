import React from 'react'
import cc from 'classcat'
import isEmpty from 'lodash.isempty'

import { Badge } from '../../badge'
import { color } from '../branding'

export type Icon = Readonly<{
  size?: number | string
  className?: string
  iconClassName?: string
  title?: string
  iconColor?: string
  badgeAriaLabel?: string
  badgeContent?: string
  isDisabled?: boolean
}>

export type IconProps = Icon &
  Readonly<{
    children: JSX.Element
    viewBox?: string
  }>

export const BaseIconDefaultProps = {
  className: '',
  iconClassName: '',
  iconColor: color.lightMidnightGreen,
  size: 24,
  title: '',
  badgeAriaLabel: '',
  badgeContent: '',
  isDisabled: false,
}

export const BaseIcon = ({
  className,
  iconClassName,
  size,
  title,
  children,
  viewBox = '0 0 24 24',
  badgeContent,
  badgeAriaLabel,
}: IconProps) => {
  const iconClasses = ['kirk-icon', iconClassName]
  if (!badgeContent) {
    // When there is no badge, the icon is not wrapped and is the root element.
    // Therefore className classes needs to be added to the icon itself.
    iconClasses.push(className)
  }

  const icon = (
    <svg
      viewBox={viewBox}
      xmlns="http://www.w3.org/2000/svg"
      className={cc(iconClasses)}
      width={size}
      height={size}
      aria-hidden={isEmpty(title)}
    >
      {title && <title>{title}</title>}
      {children}
    </svg>
  )

  if (badgeContent) {
    return (
      <div className={cc(['kirk-icon-wrapper', className])}>
        {icon}
        <Badge className="kirk-icon-badge" aria-label={badgeAriaLabel}>
          {badgeContent}
        </Badge>
      </div>
    )
  }

  return icon
}
