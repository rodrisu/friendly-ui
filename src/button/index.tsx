// Note: can't use styled components the same way we do for all other components,
// as buttons are created dynamically with React.createElement (and can be a "a", "button" or
// even every other React component => SlugLink for instance).
export { ButtonStatus, ButtonProps } from './Button'
export { StyledButton as Button } from './Button.style'
