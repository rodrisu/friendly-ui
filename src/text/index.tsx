export { TextProps, TextDisplayType, TextTagType } from './Text'
export { StyledText as Text } from './Text.style'
