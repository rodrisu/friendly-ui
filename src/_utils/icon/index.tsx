import styled from 'styled-components'

import { color, shadow } from '../branding'
import { BaseIcon } from './BaseIcon'

const StyledBaseIcon = styled(BaseIcon)`
  fill: ${props => props.iconColor ?? color.lightMidnightGreen};

  &,
  & rect,
  & circle,
  & line,
  & polyline,
  & path,
  & g {
    stroke: ${props => (props.isDisabled ? color.gray : '')};
  }

  &.kirk-icon-wrapper {
    display: inline-block;
    position: relative;
  }

  &.kirk-icon-shadowed {
    border-radius: 50%;
    box-shadow: ${shadow.icon};
  }

  & .kirk-icon-badge {
    position: absolute;
    bottom: 12px;
    left: 12px;
  }
`

export * from './BaseIcon'

export { StyledBaseIcon as BaseIcon }
