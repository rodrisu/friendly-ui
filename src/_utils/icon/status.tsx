import React, { Fragment } from 'react'

import { color } from '../branding'
import { BaseIcon } from '../icon'
import { BaseIconDefaultProps, IconProps } from '../icon/BaseIcon'

export enum status {
  ON = 'on',
  OFF = 'off',
  DEFAULT = 'default',
}

export type StatusIconProps = IconProps &
  Readonly<{
    status?: status
  }>

export type IconPropsWithStatus = Omit<StatusIconProps, 'children'>

export const StatusIcon = (props: StatusIconProps) => {
  const finalProps = {
    ...props,
    iconColor: props.status === status.ON ? color.midnightGreen : props.iconColor,
  }

  return (
    <BaseIcon {...finalProps}>
      <Fragment>
        {props.children}
        {props.status === status.OFF && (
          <g fillRule="nonzero" strokeLinecap="round" strokeLinejoin="round">
            <path stroke={color.white} d="M1,24.5 L23.5,2" />
            <path stroke={color.red} d="M1,23.5 L23.5,1" />
          </g>
        )}
      </Fragment>
    </BaseIcon>
  )
}

StatusIcon.defaultProps = {
  ...BaseIconDefaultProps,
  status: status.DEFAULT,
}
