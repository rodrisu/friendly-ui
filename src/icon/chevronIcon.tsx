import React from 'react'

import { BaseIcon, BaseIconDefaultProps, Icon } from '../_utils/icon'

export enum ChevronIconDirections {
  DOWN = 'DOWN',
  UP = 'UP',
  LEFT = 'LEFT',
}

export type ChevronIconProps = Icon &
  Readonly<{
    direction?: ChevronIconDirections
  }>

export const ChevronIcon = ({ direction, ...props }: ChevronIconProps) => (
  <BaseIcon {...props}>
    <polyline
      fill="none"
      stroke={props.iconColor}
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeMiterlimit="10"
      points="9 18 15 12 9 6"
      {...(direction === ChevronIconDirections.DOWN && { transform: 'rotate(90 12 12)' })}
      {...(direction === ChevronIconDirections.LEFT && { transform: 'rotate(180 12 12)' })}
      {...(direction === ChevronIconDirections.UP && { transform: 'rotate(-90 12 12)' })}
    />
  </BaseIcon>
)

ChevronIcon.defaultProps = {
  ...BaseIconDefaultProps,
  direction: null,
}
