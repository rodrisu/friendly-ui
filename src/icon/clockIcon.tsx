import React from 'react'

import { BaseIcon, BaseIconDefaultProps, Icon } from '../_utils/icon'

export const ClockIcon = (props: Icon) => (
  <BaseIcon {...props}>
    <g fill="none" fillRule="evenodd">
      <g stroke={props.iconColor} strokeLinecap="round" strokeLinejoin="round">
        <circle cx="12" cy="12" r="9.565" />
        <path d="M12 6.783V12h5.217" />
      </g>
    </g>
  </BaseIcon>
)

ClockIcon.defaultProps = BaseIconDefaultProps
