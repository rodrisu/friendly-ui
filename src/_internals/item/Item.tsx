import React from 'react'
import cc from 'classcat'

import { Tag as TagToken } from '../../_internals/tag'
import { color } from '../../_utils/branding'
import { A11yProps, pickA11yProps } from '../../_utils/interfaces'
import { Button } from '../../button/Button'
import { ChevronIcon } from '../../icon/chevronIcon'
import { NormalizeProps } from '../../layout/layoutNormalizer'
import { Text, TextDisplayType, TextTagType } from '../../text'

export enum ItemStatus {
  DEFAULT = 'default',
  LOADING = 'loading',
  CHECKED = 'checked',
}

export type ItemProps = A11yProps &
  NormalizeProps &
  Readonly<{
    chevron?: boolean
    className?: string
    href?: string | JSX.Element
    highlighted?: boolean
    isClickable?: boolean
    hasClickableColorRules?: boolean
    isWrappable?: boolean
    leftTitle?: React.ReactNode
    leftTitleButtonAddon?: React.ReactElement<Button>
    leftTitleDisplay?: TextDisplayType
    leftTitleColor?: string
    leftBody?: string | React.ReactNode
    leftBodyDisplay?: TextDisplayType
    leftBodyColor?: string
    leftBodyAnnotation?: React.ReactNode
    leftBodyAnnotationDisplay?: TextDisplayType
    leftBodyAnnotationColor?: string
    leftAddon?: React.ReactNode
    rightTitle?: string | JSX.Element
    rightTitleDisplay?: TextDisplayType
    rightTitleStrikeThrough?: boolean
    rightTitleAriaProps?: A11yProps
    rightTitleColor?: string
    rightTitleAddon?: JSX.Element
    rightBody?: React.ReactNode
    rightBodyDisplay?: TextDisplayType
    rightBodyStrikeThrough?: boolean
    rightBodyAriaProps?: A11yProps
    rightBodyColor?: string
    rightAddon?: React.ReactNode
    tag?: JSX.Element
    onClick?: (event: React.MouseEvent<HTMLElement>) => void
    onBlur?: (event: React.FocusEventHandler<HTMLElement>) => void
    onFocus?: (event: React.FocusEventHandler<HTMLElement>) => void
    onMouseDown?: (event: React.MouseEvent<HTMLElement>) => void
    hideHoverBackground?: boolean
    disabled?: boolean
    tagLabel?: string
  }>

export const Item = (props: ItemProps) => {
  const {
    chevron,
    className,
    href,
    onClick,
    onBlur,
    onFocus,
    onMouseDown,
    highlighted,
    isClickable,
    hasClickableColorRules = false,
    isWrappable,
    leftTitle,
    leftTitleButtonAddon,
    rightTitleStrikeThrough,
    rightTitleAriaProps,
    leftTitleDisplay = TextDisplayType.TITLE,
    leftTitleColor,
    leftBody,
    leftBodyDisplay = TextDisplayType.BODY,
    leftBodyColor,
    leftBodyAnnotation,
    leftBodyAnnotationDisplay,
    leftBodyAnnotationColor,
    leftAddon,
    rightTitle,
    rightTitleDisplay = TextDisplayType.TITLE,
    rightTitleColor,
    rightTitleAddon,
    rightBody,
    rightBodyStrikeThrough,
    rightBodyAriaProps,
    rightBodyDisplay = TextDisplayType.BODY,
    rightBodyColor,
    rightAddon,
    tag = <div />,
    hideHoverBackground = false,
    disabled = false,
    tagLabel = '',
  } = props
  const a11yAttrs = pickA11yProps<ItemProps>(props)
  let Tag = tag.type
  let tagProps = tag.props

  if (href) {
    if (typeof href !== 'string') {
      Tag = href.type
      tagProps = { ...tagProps, ...href.props }
    } else {
      Tag = 'a'
      tagProps = { href }
    }
  }

  if (Tag === 'button') {
    // The code below make sure that if we use a HTML button, a proper type='button' or
    // type='submit' is added. type='button' is the default.
    // This type=button is needed to make sure non-submit buttons are not activated when
    // submitting forms.
    // See: https://stackoverflow.com/questions/4763638/enter-triggers-button-click
    const buttonType = tag.props.type || 'button'
    tagProps = { ...tagProps, type: buttonType }
  }

  const hasRightText = rightTitle || rightBody

  const getTextColor = (textColor: string) => (disabled ? color.gray : textColor)

  const getLeftTitle = (value: React.ReactNode): React.ReactNode => {
    if (React.isValidElement(value)) {
      return leftTitle
    }

    const computeLeftTitleColor = () => {
      let colorValue: string = leftTitleColor
      // Item clickable (ItemAction, ItemActionLAbel, ItemChoice, ...): 1 line => midnightGreen, 2 lines => midnightGreen + lightMidnightGreen
      // Item non clickable (ItemInfo, ItemData ...): 1 line => lightMidnightGreen, 2 lines => midnightGreen + lightMidnightGreen
      // Item (non-)editable info:  2 lines => lightMidnightGreen + midnightGreen/blue
      if (!leftBody && !highlighted && !isClickable && !hasClickableColorRules) {
        colorValue = color.lightMidnightGreen
      }
      return getTextColor(colorValue)
    }

    return (
      <Text
        className={leftTitleButtonAddon ? 'kirk-item-title--withButtonAddon' : null}
        display={leftTitleDisplay}
        textColor={computeLeftTitleColor()}
        tag={TextTagType.SPAN}
      >
        {leftTitle}
        {/* In case of a clickable Item don't display the addon button */}
        {!href && leftTitleButtonAddon}
      </Text>
    )
  }

  return (
    <Tag
      {...tagProps}
      onClick={onClick}
      onFocus={onFocus}
      onBlur={onBlur}
      onMouseDown={onMouseDown}
      className={cc([
        'kirk-item',
        {
          'kirk-item--highlighted': highlighted,
          'kirk-item--clickable': isClickable,
          'kirk-item--wrappable': isWrappable,
          'kirk-item--hideHoverBackground': hideHoverBackground,
        },
        className,
      ])}
      {...a11yAttrs}
    >
      <span className="kirk-item-leftWrapper">
        {leftAddon && <span className="kirk-item-leftAddon">{leftAddon}</span>}
        <span className="kirk-item-leftText">
          {tagLabel && (
            <span className="kirk-item-tagToken">
              <TagToken label={tagLabel} />
            </span>
          )}
          {leftTitle && getLeftTitle(leftTitle)}
          {leftBody && (
            <Text
              className="kirk-item-body"
              display={leftBodyDisplay}
              textColor={getTextColor(leftBodyColor)}
              tag={TextTagType.SPAN}
            >
              {leftBody}
            </Text>
          )}
          {leftBodyAnnotation && (
            <Text
              className="kirk-item-body-annotation"
              display={leftBodyAnnotationDisplay}
              textColor={getTextColor(leftBodyAnnotationColor)}
              tag={TextTagType.SPAN}
            >
              {leftBodyAnnotation}
            </Text>
          )}
        </span>
      </span>
      {hasRightText && (
        <span className="kirk-item-rightText">
          {rightTitle && (
            <Text
              className={cc([
                'kirk-item-title',
                {
                  'kirk-item--strikethrough': rightTitleStrikeThrough,
                  'kirk-item-rightText--withAddon': Boolean(rightTitleAddon),
                },
              ])}
              display={rightTitleDisplay}
              textColor={getTextColor(rightTitleColor)}
              tag={TextTagType.DIV}
              {...rightTitleAriaProps}
            >
              {rightTitleAddon}
              {rightTitle}
            </Text>
          )}
          {rightBody && (
            <Text
              className={cc([
                'kirk-item-body',
                {
                  'kirk-item--strikethrough': rightBodyStrikeThrough,
                },
              ])}
              display={rightBodyDisplay}
              textColor={getTextColor(rightBodyColor)}
              tag={TextTagType.DIV}
              {...rightBodyAriaProps}
            >
              {rightBody}
            </Text>
          )}
        </span>
      )}
      {rightAddon && <span className="kirk-item-rightAddon">{rightAddon}</span>}
      {chevron && (
        <span className="kirk-item-rightAddon">
          <ChevronIcon iconColor={!isClickable ? color.gray : color.lightMidnightGreen} />
        </span>
      )}
    </Tag>
  )
}
