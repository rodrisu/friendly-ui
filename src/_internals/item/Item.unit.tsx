/* eslint-disable react/button-has-type */
import React from 'react'
import renderer from 'react-test-renderer'
import { mount, shallow } from 'enzyme'

import { render, screen } from '@testing-library/react'

import { Button } from '../../button'
import { ChevronIcon } from '../../icon/chevronIcon'
import { ClockIcon } from '../../icon/clockIcon'
import { Text } from '../../text'
import { Item as StyledItem } from './index'
import { Item } from './Item'

describe('Item', () => {
  it('Should not have changed', () => {
    const item = renderer
      .create(
        <StyledItem
          className="custom"
          leftTitle="Test"
          leftBody="Test"
          rightAddon={<ClockIcon />}
        />,
      )
      .toJSON()
    expect(item).toMatchSnapshot()
  })

  it('Should be displayed in default state without props', () => {
    const wrapper = shallow(<Item />)
    expect(wrapper.hasClass('kirk-item')).toBe(true)
    expect(wrapper.type()).toBe('div')
  })

  it('Should accept a custom `className`', () => {
    const customClassName = 'custom'
    const wrapper = shallow(<Item className={customClassName} />)
    expect(wrapper.hasClass(customClassName)).toBe(true)
  })

  it('Should be highlighted', () => {
    const wrapper = shallow(<Item highlighted />)
    expect(wrapper.hasClass('kirk-item--highlighted')).toBe(true)
  })

  it("Shouldn't be wrappable by default", () => {
    render(<Item leftTitle="Test" tag={<button />} />)
    expect(screen.getByRole('button', { name: 'Test' })).not.toHaveClass('kirk-item--wrappable')
  })

  it('Should be wrappable', () => {
    render(<Item leftTitle="Test" tag={<button />} isWrappable />)
    expect(screen.getByRole('button', { name: 'Test' })).toHaveClass('kirk-item--wrappable')
  })

  it('Should deactivate background hover color properly', () => {
    const wrapperWithHoverBackground = shallow(<Item />)
    expect(wrapperWithHoverBackground.hasClass('kirk-item--hideHoverBackground')).toBe(false)

    const wrapperWithoutHoverBackground = shallow(<Item hideHoverBackground />)
    expect(wrapperWithoutHoverBackground.hasClass('kirk-item--hideHoverBackground')).toBe(true)
  })

  it('Should trigger click on button info', () => {
    const onButtonClick = jest.fn()
    const wrapper = mount(
      <Item
        leftTitle="Left title"
        leftTitleButtonAddon={<Button onClick={onButtonClick}>More info</Button>}
      />,
    )
    wrapper.find('.kirk-item-title--withButtonAddon button').simulate('click')
    expect(onButtonClick).toHaveBeenCalledTimes(1)
  })

  it('Should use correct button type for normal buttons', () => {
    const wrapper = mount(<Item tag={<button />} />)
    expect(wrapper.find('button[type="button"]').exists()).toBe(true)
  })

  it('Should not add type=button to anchors', () => {
    // The Item uses a <button> tag and href. Because of the href using an anchor, the item will be
    // transformed into an HTML anchor, not HTML button.
    const wrapper = mount(<Item tag={<button />} href={<a href="slug_link" />} />)

    // Verify that there is a proper generated anchor element.
    expect(wrapper.find('button[type="button"]').exists()).toBe(false)
    expect(wrapper.find('a[type="button"]').exists()).toBe(false)
    expect(wrapper.find('a').exists()).toBe(true)
  })

  it('Should use correct button type for submit buttons', () => {
    const wrapper = mount(<Item tag={<button type="submit" />} />)
    expect(wrapper.find('button[type="submit"]').exists()).toBe(true)
  })

  it("Should't display left button addon if no title", () => {
    const wrapper = mount(
      <Item leftTitleButtonAddon={<Button onClick={() => {}}>More info</Button>} />,
    )
    expect(wrapper.find('.kirk-item-title--withButtonAddon button').exists()).toBe(false)
  })

  it("Should't display left button addon if Item is clickable", () => {
    const wrapper = mount(
      <Item
        href={<a href="#" />}
        leftTitle="Left title"
        leftTitleButtonAddon={<Button onClick={() => {}}>More info</Button>}
      />,
    )
    expect(wrapper.find('.kirk-item-title--withButtonAddon button').exists()).toBe(false)
  })

  it('Should accept a custom `tag`', () => {
    const wrapper = shallow(<Item tag={<li />} />)
    expect(wrapper.type()).toBe('li')
  })

  it('Should not return an href prop when no href is pass to the item', () => {
    const wrapper = shallow(<Item />)
    expect(wrapper.prop('href')).toBeUndefined()
  })

  it('Should return a tag A when href is a string', () => {
    const wrapper = shallow(<Item href="#" />)
    expect(wrapper.type()).toBe('a')
    expect(wrapper.prop('href')).toBe('#')
  })

  it('Should return a tag of a with its href when href is a component a', () => {
    const wrapper = shallow(<Item href={<a href="#" />} />)
    expect(wrapper.type()).toEqual('a')
    expect(wrapper.prop('href')).toEqual('#')
  })
  it('Should return a <a /> tag without extra attribrutes when href is a string', () => {
    const wrapper = shallow(<Item href="#" tag={<button type="button" />} />)
    expect(wrapper.type()).toEqual('a')
    expect(wrapper.prop('type')).toBe(undefined)
  })

  it('Should display a left add-on', () => {
    const wrapper = shallow(<Item leftAddon={<ClockIcon />} />)
    expect(wrapper.find(ClockIcon).exists()).toBe(true)
  })

  it('Should display a right add-on', () => {
    const wrapper = shallow(<Item rightAddon={<ClockIcon />} />)
    expect(wrapper.find(ClockIcon).exists()).toBe(true)
  })

  it('Should display a chevron', () => {
    const wrapper = shallow(<Item chevron />)
    expect(wrapper.find(ChevronIcon).exists()).toBe(true)
  })

  it('Should display left body annotation', () => {
    const wrapper = mount(<Item leftBodyAnnotation="Left body annotation label" />)
    expect(wrapper.find('.kirk-item-body-annotation').exists()).toBe(true)
  })

  it('Should display left body', () => {
    const wrapper = mount(<Item leftBody="Left body label" />)
    expect(wrapper.find('.kirk-item-body').exists()).toBe(true)
  })

  it('Should display strickthrough right title', () => {
    const wrapper = mount(<Item rightTitle="Right title" rightTitleStrikeThrough />)
    expect(wrapper.find('.kirk-item--strikethrough').exists()).toBe(true)
  })

  it("Shouldn't display strickthrough right title", () => {
    const wrapper = mount(<Item rightTitleStrikeThrough />)
    expect(wrapper.find('.kirk-item--strikethrough').exists()).toBe(false)
  })

  it('Should display aria-label on right title', () => {
    const wrapper = mount(
      <Item
        rightTitle="Right title"
        rightTitleAriaProps={{ 'aria-label': 'Right title aria-label' }}
      />,
    )
    expect(wrapper.find(Text).prop('aria-label')).toBe('Right title aria-label')
  })
})
