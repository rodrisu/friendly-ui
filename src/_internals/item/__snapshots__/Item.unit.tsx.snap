// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Should not have changed 1`] = `
.c2 {
  fill: #708C91;
}

.c2.kirk-icon-wrapper {
  display: inline-block;
  position: relative;
}

.c2.kirk-icon-shadowed {
  border-radius: 50%;
  box-shadow: 0 0 2px rgba(0,0,0,.3);
}

.c2 .kirk-icon-badge {
  position: absolute;
  bottom: 12px;
  left: 12px;
}

.c1 {
  margin: 0;
  font-weight: 400;
}

.c1.kirk-text-button {
  color: #054752;
  font-size: 16px;
  line-height: 20px;
}

.c1.kirk-text-body,
.c1.kirk-text-bodyStrong {
  color: #708C91;
  font-size: 16px;
  line-height: 20px;
}

.c1.kirk-text-caption {
  color: #708C91;
  font-size: 13px;
  line-height: 16px;
}

.c1.kirk-text-title,
.c1.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c1.kirk-text-display2 {
  color: #054752;
  font-size: 82px;
  line-height: 82px;
  font-weight: 500;
}

.c1.kirk-text-display1 {
  color: #054752;
  font-size: 30px;
  line-height: 1.06;
  font-weight: 500;
}

.c1.kirk-text-subheader,
.c1.kirk-text-subheaderStrong {
  color: #054752;
  font-size: 22px;
  line-height: 24px;
}

.c1.kirk-text-title,
.c1.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c1.kirk-text-bodyStrong,
.c1.kirk-text-subheaderStrong,
.c1.kirk-text-titleStrong {
  font-weight: 500;
}

.c0 {
  padding-left: 24px !important;
  padding-right: 24px !important;
  margin-left: 0;
  margin-right: 0;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 0;
  background: none;
  white-space: pre-line;
}

.c0.kirk-item--clickable:hover {
  cursor: pointer;
  background: #EDEDED;
}

.c0.kirk-item--hideHoverBackground.kirk-item--clickable:hover {
  background: none;
}

.c0.kirk-item--wrappable {
  -webkit-flex-flow: wrap;
  -ms-flex-flow: wrap;
  flex-flow: wrap;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c0.kirk-item--wrappable .kirk-item-leftWrapper {
  padding-top: 8px;
  padding-bottom: 8px;
}

.c0.kirk-item--wrappable .kirk-item-leftText {
  margin-right: 24px;
}

.c0.kirk-item--wrappable .kirk-item-rightText {
  margin-left: 0;
  margin-top: 8px;
  margin-bottom: 8px;
  text-align: left;
}

.c0 .kirk-item-tagToken {
  display: block;
  padding-bottom: 4px;
}

.c0:disabled {
  cursor: default;
}

a.c0 {
  background: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button.c0 {
  cursor: pointer;
  text-align: left;
  width: 100%;
  font-family: inherit;
  background-color: transparent;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button:focus:not(.focus-visible).c0 {
  outline: none;
}

.c0 .kirk-item-leftWrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .kirk-item-leftText {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 .kirk-item-rightText {
  margin-left: 16px;
  text-align: right;
}

.c0 .kirk-text {
  display: block;
}

.c0 .kirk-item-title--withBody {
  margin-bottom: 4px;
}

.c0 .kirk-item--strikethrough {
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

.c0 .kirk-item-title--withButtonAddon {
  position: relative;
}

.c0 .kirk-item-leftAddon,
.c0 .kirk-item-rightAddon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-width: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .kirk-item-leftAddon {
  margin-right: 16px;
}

.c0 .kirk-item-rightAddon {
  margin-left: 16px;
}

.c0 .kirk-item-rightText--withAddon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .kirk-item-rightText--withAddon svg {
  margin-right: 8px;
}

.c0.kirk-item--highlighted .kirk-text-title {
  color: #00AFF5;
}

.c0 .kirk-item-title--withButtonAddon button {
  position: absolute;
  width: 24px;
  height: 24px;
  margin-left: 4px;
  bottom: 0;
}

<div
  className="kirk-item c0 custom"
>
  <span
    className="kirk-item-leftWrapper"
  >
    <span
      className="kirk-item-leftText"
    >
      
      <span
        className="kirk-text kirk-text-title c1"
      >
        Test
      </span>
      <span
        className="kirk-text kirk-text-body c1 kirk-item-body"
      >
        Test
      </span>
    </span>
  </span>
  <span
    className="kirk-item-rightAddon"
  >
    <svg
      aria-hidden={true}
      className="kirk-icon c2"
      height={24}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      
      <g
        fill="none"
        fillRule="evenodd"
      >
        <g
          stroke="#708C91"
          strokeLinecap="round"
          strokeLinejoin="round"
        >
          <circle
            cx="12"
            cy="12"
            r="9.565"
          />
          <path
            d="M12 6.783V12h5.217"
          />
        </g>
      </g>
    </svg>
  </span>
</div>
`;
